### PR TITLE
Feature/fingerbank cache rework

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Use the Fingerbank cache instead of caching its result
# Impacts

Fingerbank calls
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Fingerbank caching is now done through the Fingerbank library

### CODE REVIEW

File: lib/pf/fingerbank.pm [hunk 1]
```diff
@@ -100,14 +99,9 @@ sub _query {
...
-        sub {
-            $logger->debug("Fingerbank result not in cache (either not present or expired). Querying Fingerbank for result");
-            my $fingerbank = fingerbank::Query->new;
-            return $fingerbank->match($args);
-        },
-        {expires_in => FINGERBANK_CACHE_EXPIRE}
-    );
+    my $cache = pf::CHI->new( namespace => 'fingerbank', expires_in => FINGERBANK_CACHE_EXPIRE );
+    my $fingerbank = fingerbank::Query->new(cache => $cache);
+    return $fingerbank->match($args);
```
Comment (fdurand) [inline lib/pf/fingerbank.pm:95]: isn't it $cached_args ?

Comment (julsemaan) [inline lib/pf/fingerbank.pm:95]: No and I actually need to remove cached_args from what I see

File: lib/pf/fingerbank.pm [hunk 2]
```diff
@@ -100,14 +99,9 @@ sub _query {
...
 
-    return $cache->compute_with_undef($cached_args, 
-        sub {
-            $logger->debug("Fingerbank result not in cache (either not present or expired). Querying Fingerbank for result");
-            my $fingerbank = fingerbank::Query->new;
-            return $fingerbank->match($args);
-        },
-        {expires_in => FINGERBANK_CACHE_EXPIRE}
-    );
+    my $cache = pf::CHI->new( namespace => 'fingerbank', expires_in => FINGERBANK_CACHE_EXPIRE );
```
Comment (jrouzierinverse) [inline lib/pf/fingerbank.pm:93]: put expires_in in the conf/chi.conf.example

Comment (julsemaan) [inline lib/pf/fingerbank.pm:93]: We can't now as this will hit the maintenance branch and chi.conf will not be upgraded through it.

If we don't put it here and it's not in chi.conf then it will never expire.

We should create a chi.conf.defaults to handle these cases, as I agree with you, its not clean to have it here now

Comment (jrouzierinverse) [inline lib/pf/fingerbank.pm:93]: Done #1041

Comment (julsemaan) [inline lib/pf/fingerbank.pm:93]: Perfect.

I added a note to #1041 to remove this once its merged (as I doubt it will be in maintenance)

File: lib/pf/fingerbank.pm (hunk 1 above)
Comment (julsemaan) [inline lib/pf/fingerbank.pm:95]: Fixed, cached_args removed

File: lib/pf/fingerbank.pm (hunk 2 above)
Comment (rouzier) [inline lib/pf/fingerbank.pm:93]: #1041 could also be added to maintenance.

Comment (julsemaan) [inline lib/pf/fingerbank.pm:93]: maintenance is usually only for bug fixes or to resolve issue that cannot wait for the next release.

Introducing a more complex change like reworking the CHI config seems out of scope for the maintenance

Comment (dwlfrth) [inline lib/pf/fingerbank.pm:93]: I agree with @julsemaan

**Comment by julsemaan** [comment]
Ready to merge so it can be part of 5.5.2 today ?
